pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -697,10 +708,28 @@ def unique(l):
                     pdd[d][qbn] = pdd['split_data_dict'][qbn][p][ind][d]
                 self.measurement_strings[qbn] += f' ({p}: {v})'
 
-    def get_cal_data_points(self):
+    def get_num_cal_points(self):
+        if self.cal_states_dict is None:
+            self.cal_states_dict = {}
         self.num_cal_points = np.array(list(
             self.cal_states_dict.values())).flatten().size
 
+        spd = self.proc_data_dict['sweep_points_dict']
+        num_sp = len(spd[list(spd)[0]]['sweep_points'])
+        mrpq = self.proc_data_dict['meas_results_per_qb']
+        mrpq_raw_dict = mrpq[list(mrpq)[0]]
+        num_data_points = len(mrpq_raw_dict[list(mrpq_raw_dict)[0]])
+        if self.num_cal_points == 0 and num_data_points != num_sp:
```

Comment:
When is this condition expected to be True? Please add a comment

Reply:
Comment added. It when no calibration points information was given (cal_points in options dict or metadata is None or an empty CalibrationPoints instance), but the measurement did contain calibration segments. We want to exclude the data corresponding to them from the fit.